### PR TITLE
Firtst thoughts about guide for reviewers #2688

### LINKED ISSUE #2688: Create guidelines for reviewers
### Situation
Currently it is sometimes not clear what is expected from a pull request review.

### Wanted
Document steps.

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #2688

### CODE REVIEW

File: sechub-doc/src/docs/asciidoc/documents/other/review-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,487 @@
+:toc: left
+== Reviewers guide
+
+At {sechub} we have some quality aspects for pull requests.
```
Comment (Abdullah-Benomar-Shahen) [inline sechub-doc/src/docs/asciidoc/documents/other/review-guide.adoc:4]: Once cloud also add a PR Template on the Repo level which includes some checkboxes to ensure the Author thought about some crucial points (Based on this Doc).

I have done it my current projects at work and it really helped me reduce some noise around Code Reviews and PRs.

See: [PR Template Guide](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)

**Comment by de-jcup** [comment]
@Abdullah-Benomar-Shahen : first of all: thank you for your ideas and your suggestions. This was only a first draft, means only an idea.

IMHO it was too much content from myself about a PR - normally just a "rethink if all parts are done and if there is something missing" could be enough.

On the other hand your suggestions are very wellcome .

Currently I have many other issues so this will still need some time.

File: sechub-doc/src/docs/asciidoc/documents/other/review-guide.adoc (hunk 1 above)
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/documents/other/review-guide.adoc:4]: good idea, thank you

File: sechub-doc/src/docs/asciidoc/documents/other/review-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+=== Code
+
+The main purpose here is
+
+- readability (others shall understand the code easily)
+- maintainability (it shall be clear how the code works and how a change can be made)
+- security aspects
+- no typical flaws
+- common style
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/documents/other/review-guide.adoc:57]: We currently use spotless to check for correct formatting.

But we could integrate other tools like PMD,Checkstyle or SpotBugs in future - good point.

I was thinking about using arch unit as well in the past, but I decided to simply write some extra unit tests to test some parts (e.g. domains cannot communicate directly with each other to keep DDD working). But maybe arch unit will come at some point of time.

**Comment by de-jcup** [comment]
In the mean time we have integrated ArchUnit for architectural checks and we will keep on creating new rules etc.

I will close this issue but keep the ideas about PMD, Template guide etc. in mind.

**Pull request closed without merging**